src/files/file-preview/FilePreview.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import isBinary from 'is-binary'
+import fileExtension from 'file-extension'
+import typeFromExt from '../type-from-ext'
+
+class FilesPreview extends React.Component {
+  static propTypes = {
+    stats: PropTypes.object.isRequired,
+    gatewayUrl: PropTypes.string.isRequired,
+    read: PropTypes.func.isRequired,
+    content: PropTypes.object
+  }
+
+  state = {
+    content: null
+  }
+
+  loadContent () {
+    this.props.read()
+      .then(buf => {
+        this.setState({ content: buf.toString('utf-8') })
+      })
+  }
+
+  render () {
+    const {stats, gatewayUrl} = this.props
+
+    const type = typeFromExt(stats.name)
+    const src = `${gatewayUrl}/ipfs/${stats.hash}`
+    const ext = fileExtension(stats.name)
+    const className = 'w-100 bg-snow-muted pa2 br2'
+    switch (type) {
+      case 'audio':
+        return (
+          <audio width='100%' controls>
+            <source src={src} type={`audio/${ext}`} />
+          </audio>
+        )
+      case 'pdf':
+        return (
+          <object width='100%' height='500px' data={src} type='application/pdf'>
+            If you're seeing this, is because your browser doesn't support previwing
+            PDF files.
+          </object>
+        )
+      case 'video':
+        console.log('Ivdeo')
+        return (
+          <video controls className={className}>
+            <source src={src} type={`video/${ext}`} />
+          </video>
+        )
+      case 'image':
+        return <img className={className} alt={stats.name} src={src} />
+      default:
+        const cantPreview = (
+          <div>
+            Sorry, we can not preview this file..
```

Comment:
There are going to be a bunch of media types we can't preview in a browser... @akrych can you add designs for "preview page for file types where we can't show a preview of the content". We can still ask IPFS for a file size at least, but that isn't much.

More generally, I think that doing a good job of showing a preview for all types of files is a lot of work. Like, many files will say they are are "video" but that doesn't mean we have the codecs to play them. Same for audio. 

We could do extra work and unpack zip files and show their structure but then we've got tar,7zip,rar etc which would each need custom previews.

Then there is all the text flavours. Do we go full github and try and render all of them too? W

We need to at least catch all the scenarios where the current browser can't preview a file that might normally be playable. Showing a video element, that just shows a gray box is a bad user experience, and worse than just saying we can't preview it.

We could preview files in browser tabs for now, and let the browser decide what to do with it. It's not great, but it avoids a load of edge cases for until we're ready to do all the work that a robust, general purpose preview page will need.

@lidel @alanshaw @hacdias what are your thoughts on this?

Reply:
I agree with @olizilla , not sure if we want to go into the rabbit hole of adding file previews of too many formats.  It will be time consuming and it *could easily be added later via PRs* (great opportunity for contributors). That being said, two additional ideas below.

#### Improving UX without real previews

It may have more sense to rename it to "File Details"  (as it does not imply preview is possible).

Even if we can't play a `.mov` video or open a `.docx` document, we can visually communicate to user if a file is a video, document, archive.

To roughly determine if preview can be rendered, we could do proper mime-type sniffing via something like [file-type](https://github.com/sindresorhus/file-type) (it only needs the first 4100 bytes of every file).

In absence of  real preview for specific format,  the screen could display file size, detected mime-type with some human-readable description of the file type, a different Icon for popular mime-types (or just generic image/sound/video/document/archive/unknown) and a <kbd>Download</kbd> button to enable user to open file in third-party handler provided by OS.

#### Previews for web-safe formats

If we want to have minimal support for previews we should aim at low hanging fruits: images, text files and basic audio and video player for [formats supported by the browser](https://developer.mozilla.org/en-US/docs/Web/HTML/Supported_media_formats) with a fallback <kbd>Download</kbd> button if it can't be played for some reason.